src/Microsoft.AspNet.Routing/VirtualPathData.cs

```diff
@@ -57,23 +56,23 @@ public VirtualPathData(IRouter router, string virtualPath)
 
             Router = router;
             VirtualPath = virtualPath;
-
-            _dataTokens = new RouteValueDictionary();
-            if (dataTokens != null)
-            {
-                foreach (var dataToken in dataTokens)
-                {
-                    _dataTokens.Add(dataToken.Key, dataToken.Value);
-                }
-            }
+            _dataTokens = dataTokens == null ? null : new RouteValueDictionary(dataTokens);
         }
 
         /// <summary>
         /// Gets the collection of custom values for the <see cref="Router"/>.
         /// </summary>
-        public IDictionary<string, object> DataTokens
+        public RouteValueDictionary DataTokens
         {
-            get { return _dataTokens; }
+            get
+            {
+                if (_dataTokens == null)
```

Comment:
IE was having issues and by the time I got the PR back up, you had merged. :smile: